docs/configuration.md

```diff
@@ -0,0 +1,124 @@
+# Configuration settings
+
+This document provides an overview of configuration settings for increasing hub performance.
+
+## Culler settings
+There are two mechanisms for controlling the culling of servers and users. One is a
+process managed by the hub which will periodically cull users and servers. The other
+is a setting which will allow servers to delete themselves after a period of inactivity.
+
+### Frequency
+By default the culler runs every 10 minutes. With a more aggressive setting for the notebook
+idle timeout the hub-managed culler can be run less frequently.
+
+### Concurrency limit
+By default the culler has a concurrency limit of 10. This means it will make up to 10
+concurrent API calls. When deleting a large number of users that can generate a high load
+on the hub. Setting this to `1` helps to reduce load on the hub.
+
+### Timeout
+The timeout controls how long a server can be idle before being deleted. Because the servers
+will aggressively cull themselves this value can be set very high.
+
+These can be all configured in the `cull` section of `values.yaml`:
+```yaml
+cull:
+  timeout: 432000 # 5 days
+  every: 3600 # Run once an hour instead of every 10 minutes
+  concurrency: 1
+```
+
+### Notebook culler
+How long a server can be idle before culling itself is controlled by the `IDLE_TIMEOUT`
+environment variable. This is configured in `singleuser.extraEnv` in `values.yaml`.
+
+```yaml
+singleuser:
+  extraEnv:
+    IDLE_TIMEOUT: 14400 # 4 hours
```

Comment:
This is actually special sauce in our user image since we have per-environment values files but a single user image so we can control the timeout per environment in our helm chart values files but this `IDLE_TIMEOUT` environment variable is specific to our user image.

So to make this generic it would probably be better to just mention these user notebook options and what we set them to:

- `c.NotebookApp.shutdown_no_activity_timeout`
- `c.MappingKernelManager.cull_idle_timeout`

We should probably also use 3600 as the example in here since that's what we use in production (notebooks cull themselves after 1 hour of idle kernel).

Reply:
I didn't see anything else in the various jupyterhub repos so I assumed that was it. I see now that that setting is used in jupyter/notebook. I'll update this section.